src/app/components/turnos/gestor-agendas/operaciones-turnos/liberar-turno.component.ts

```diff
@@ -30,33 +32,45 @@ export class LiberarTurnoComponent implements OnInit {
     public motivoLiberacionSelect;
     public otroMotivoLiberacion;
 
-    constructor(public plex: Plex, public listaEsperaService: ListaEsperaService, public serviceAgenda: AgendaService) { }
+    constructor(public plex: Plex,
+        public listaEsperaService: ListaEsperaService,
+        public serviceAgenda: AgendaService,
+        public prestacionesService: PrestacionesService) { }
 
     ngOnInit() {
         this.turnos = this.turnosSeleccionados;
         this.motivoLiberacionSelect = this.motivoLiberacion[1];
     }
 
     liberarTurno() {
-        if (this.motivoLiberacionSelect) {
-            let patch = {
-                op: 'liberarTurno',
-                turnos: this.turnos.map(resultado => resultado._id),
-                observaciones: this.motivoLiberacionSelect.nombre === 'Otro' ? this.otroMotivoLiberacion : this.motivoLiberacionSelect.nombre
-            };
-            let mensaje = this.turnos.length === 1 ? 'El turno seleccionado fue liberado' : 'Los turnos seleccionados fueron liberados';
-
-            this.serviceAgenda.patch(this.agenda.id, patch).subscribe(resultado => {
-                this.plex.toast('success', mensaje, 'Liberar turno', 4000);
-                this.saveLiberarTurno.emit(this.agenda);
-            },
-            err => {
-                if (err) {
-                    this.plex.info('warning', 'Turno en ejecución', 'Error');
-                    this.cancelaLiberarTurno.emit(true);
+        this.prestacionesService.get({ turnos: this.turnos[0].id }).subscribe((prestacion: any) => {
+            if (prestacion?.length) {
+                const estadoPrestacion = prestacion[0].estados[prestacion[0].estados.length - 1].tipo;
```

Comment:
Evaluar directamente el atributo `estadoActual.tipo`
```suggestion
                const estadoPrestacion = prestacion[0].estadoActual.tipo;
```